packages/aws-cdk/lib/api/hotswap/appsync-mapping-templates.ts

```diff
@@ -0,0 +1,149 @@
+  ) {
+    this.resourceNames = [`AppSync resolver '${resolverUpdateRequest.name}'`];
+  }
+
+  public async apply(sdk: ISDK): Promise<any> {
+    return sdk.appsync().updateResolver(this.resolverUpdateRequest.request).promise();
+  }
+}
+
+async function isHotswappableAppSyncFunctionChange(
+  logicalId: string, change: HotswappableChangeCandidate, evaluateCfnTemplate: EvaluateCloudFormationTemplate,
+): Promise<ChangeHotswapResult> {
+  if (change.newValue.Type !== 'AWS::AppSync::FunctionConfiguration') {
+    return ChangeHotswapImpact.REQUIRES_FULL_DEPLOYMENT;
+  }
+
+  const functionName = await evaluateCfnTemplate.establishResourcePhysicalName(logicalId, change.newValue.Properties?.Name);
+  if (!functionName) {
+    return ChangeHotswapImpact.REQUIRES_FULL_DEPLOYMENT;
+  }
+
+  const updateRequest: FunctionUpdateRequest = {
+    name: functionName,
+    request: {
+      apiId: await evaluateCfnTemplate.evaluateCfnExpression(change.newValue.Properties?.ApiId),
+      functionVersion: change.newValue.Properties?.FunctionVersion,
+      dataSourceName: change.newValue.Properties?.DataSourceName,
+      requestMappingTemplate: await evaluateCfnTemplate.evaluateCfnExpression(change.newValue.Properties?.RequestMappingTemplate),
+      responseMappingTemplate: await evaluateCfnTemplate.evaluateCfnExpression(change.newValue.Properties?.ResponseMappingTemplate),
+      name: functionName,
+    },
+  };
+
+  for (const updatedPropName in change.propertyUpdates) {
+    const updatedProp = change.propertyUpdates[updatedPropName];
+
+    switch (updatedPropName) {
+      case 'RequestMappingTemplate':
+        updateRequest.request.requestMappingTemplate = await evaluateCfnTemplate.evaluateCfnExpression(updatedProp.newValue);
+        break;
+      case 'ResponseMappingTemplate':
+        updateRequest.request.responseMappingTemplate = await evaluateCfnTemplate.evaluateCfnExpression(updatedProp.newValue);
+        break;
+      default:
+        return ChangeHotswapImpact.REQUIRES_FULL_DEPLOYMENT;
+    }
+  }
+
+  return new FunctionHotswapOperation(updateRequest);
+}
+
+interface FunctionUpdateRequest {
+  name: string
+  request: Omit<AWS.AppSync.UpdateFunctionRequest, 'functionId'>
+}
+
+class FunctionHotswapOperation implements HotswapOperation {
+  public readonly service = 'appsync'
+  public readonly resourceNames: string[];
+
+  constructor(
+    private readonly functionUpdateRequest: FunctionUpdateRequest,
+  ) {
+    this.resourceNames = [`AppSync function '${functionUpdateRequest.name}'`];
+  }
+
+  public async apply(sdk: ISDK): Promise<any> {
+    const { functions } = await sdk.appsync().listFunctions({ apiId: this.functionUpdateRequest.request.apiId }).promise();
+    const { functionId } = functions?.find(fn => fn.name === this.functionUpdateRequest.name) || {};
+    if (!functionId) {
+      throw new Error('Unable to find matching function for hotswap operation.');
+    }
```

Comment:
@skinny85 it feels bad to get this far and bail; I realize this leaves the app in a bad state as it's too late to do a full deploy. I have to have the `functionId`, though, and it doesn't seem to be available elsewhere. If we passed the sdk into the earlier checks I could handle it beforehand; lmk your thoughts!